src/main/java/dan200/computercraft/ComputerCraft.java

```diff
@@ -252,10 +256,34 @@ public void preInit( FMLPreInitializationEvent event )
         Config.config.load();
 
         Config.http_enable = Config.config.get( Configuration.CATEGORY_GENERAL, "http_enable", http_enable );
-        Config.http_enable.setComment( "Enable the \"http\" API on Computers (see \"http_whitelist\" for more fine grained control than this)" );
+        Config.http_enable.setComment( "Enable the \"http\" API on Computers (see \"http_whitelist\" and \"http_blacklist\" for more fine grained control than this)" );
 
-        Config.http_whitelist = Config.config.get( Configuration.CATEGORY_GENERAL, "http_whitelist", http_whitelist );
-        Config.http_whitelist.setComment( "A semicolon limited list of wildcards for domains that can be accessed through the \"http\" API on Computers. Set this to \"*\" to access to the entire internet. Example: \"*.pastebin.com;*.github.com;*.computercraft.info\" will restrict access to just those 3 domains." );
+        {
+            ConfigCategory category = Config.config.getCategory( Configuration.CATEGORY_GENERAL );
+            Property currentProperty = category.get( "http_whitelist" );
+            if( currentProperty != null && !currentProperty.isList() ) category.remove( "http_whitelist" );
+
+            Config.http_whitelist = Config.config.get( Configuration.CATEGORY_GENERAL, "http_whitelist", new String[] { "*" } );
+
+            if( currentProperty != null && !currentProperty.isList() )
+            {
+                Config.http_whitelist.setValues( currentProperty.getString().split( ";" ) );
+            }
+        }
+        Config.http_whitelist.setComment( "A list of wildcards for domains or IP ranges that can be accessed through the \"http\" API on Computers.\n" +
+            "Set this to \"*\" to access to the entire internet. Example: \"*.pastebin.com\" will restrict access to just subdomains of pastebin.com.\n" +
+            "You can use domain names (\"pastebin.com\"), wilcards (\"*.pastebin.com\") or CIDR notation (\"127.0.0.0/8\")." );
+
+        Config.http_blacklist = Config.config.get( Configuration.CATEGORY_GENERAL, "http_blacklist", new String[] {
```

Comment:
Yeah, stick them up top somewhere, and construct the http_blacklist global using them.